### PR TITLE
Enable SKUs on Linux by default

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves https://github.com/brave/brave-browser/issues/32803

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Purchase a Brave News subscription on account.bravesoftware.com.  The browser should not crash.

### CODE REVIEW

File: browser/browser_context_keyed_service_factories.cc [hunk 1]
```diff
@@ -174,9 +178,11 @@ void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
   EphemeralStorageServiceFactory::GetInstance();
   PermissionLifetimeManagerFactory::GetInstance();
 #if BUILDFLAG(ENABLE_BRAVE_VPN)
-  skus::SkusServiceFactory::GetInstance();
   brave_vpn::BraveVpnServiceFactory::GetInstance();
 #endif
+#if BUILDFLAG(ENABLE_BRAVE_VPN) || BUILDFLAG(ENABLE_AI_CHAT)
```
Comment (bsclifton) [inline browser/browser_context_keyed_service_factories.cc:183]: `SkusServiceFactory` needs to be initialized BEFORE `BraveVpnServiceFactory`. This is changing the order. Can you move this check for `#if BUILDFLAG(ENABLE_BRAVE_VPN) || BUILDFLAG(ENABLE_AI_CHAT)` above the VPN check?

Comment (bsclifton) [reply browser/browser_context_keyed_service_factories.cc:180]: Perfect - thanks for updating 🙂👍

**Review by bsclifton — APPROVED** [review]
LGTM! 😄👍

**Comment by nvonpentz** [comment]
@fmarier @wknapik I requested your review because of your involvement in https://github.com/brave/brave-browser/issues/23140. I am running Ubuntu and didn't have issues with crashes after these changes.

**Comment by nvonpentz** [comment]
Thanks @wknapik 
> I tested logging into my premium account on latest nightly, without your change, and the browser didn't crash for me.

Right, I think current `master` doesn't have crashes, but is also missing support for SKUs (because it was causing crashes). I verified the lack of support by going to `brave://skus-internals/`, which didn't load until I made this change.

Rebasing.

**Comment by bsclifton** [comment]
Here's the issue I captured for the crash on Linux:
https://github.com/brave/brave-browser/issues/23348

From my notes there:
> I've only been able to reproduce with official build (Nightly for example)

If we wanted to be extra safe, we can do a release build locally WITH PGO profiles. If this doesn't have a crash when logging into account.brave.com then I think we're good to go 🙂 We could always test with Nightly too

**Comment by nvonpentz** [comment]
Based on my own testing locally I am comfortable merging as is and continue testing on Nightly.

**Comment by bsclifton** [comment]
Let's do it 😄 (once CI passes, of course)

**Comment by stephendonner** [comment]
> ## Test Plan:
> Purchase a Brave News subscription on account.bravesoftware.com. The browser should not crash.

@nvonpentz you mean `Brave Talk`, `Brave Search`, or `Brave Leo`, right?

(And we should definitely test w/Leo.)

**Pull request merged**